src/server/game/Server/WorldSession.cpp

```diff
@@ -637,6 +637,9 @@ void WorldSession::LogoutPlayer(bool save)
                     sScriptMgr->OnBattlegroundDesertion(_player, BG_DESERTION_TYPE_INVITE_LOGOUT);
                 }
 
+                if (bgQueueTypeId >= BATTLEGROUND_QUEUE_2v2 && bgQueueTypeId < MAX_BATTLEGROUND_QUEUE_TYPES && _player->IsInvitedForBattlegroundQueueType(bgQueueTypeId))
+                    sScriptMgr->OnBattlegroundDesertion(_player, ARENA_DESERTION_TYPE_INVITE_LOGOUT);
+
```

Comment:
this relies on the order of entries in `BattlegroundQueueTypeId` to detect if it's an arena match? is there probably a "nicer" way to decide if it's arena? also `BattlegroundQueueTypeId` currently has a big gap between `BATTLEGROUND_QUEUE_5v5`  and `MAX_BATTLEGROUND_QUEUE_TYPES`, adding entries there would also trigger this call.
but maybe it's just irrelevant

Reply:
it's done on purpose, so we can add more custom entries between those two values